lib/galaxy/jobs/runners/drmaauniva.py

```diff
@@ -0,0 +1,640 @@
+"""
+Kind of fail save job control via the DRMAA API.
+Tested for UNIVA grid engine
+"""
+import logging
+import os
+import re
+import signal
+import subprocess
+import time
+
+# from galaxy import model
+from galaxy.jobs.runners.drmaa import DRMAAJobRunner
+
+
+log = logging.getLogger( __name__ )
+
+__all__ = ( 'DRMAAUnivaJobRunner', )
+
+MEMORY_LIMIT_SCAN_SIZE = 1024 * 1024  # 1MB
+
+"""
+TODO
+find out what it means if there is no error code file (.ec)
+can this also be used
+
+tool wrapper:
+- A tool that is the bash script itself.
+- bash script (create long array, dd into variable)
+- C
+- C++
+- python
+- java
+- perl
+"""
+
+
+class DRMAAWaitUnusable( Exception ):
+    pass
+
+
+class DRMAAUnivaJobRunner( DRMAAJobRunner ):
```

Comment:
For consistency with the `SlurmJobRunner` runner, I'd go with just `UnivaJobRunner`.

Reply:
has been replaced in the whole file